lib/workers/repository/extract/extract-fingerprint-config.ts

```diff
@@ -0,0 +1,83 @@
+import { getManagerConfig, mergeChildConfig } from '../../../config';
+import type {
+  RegexManagerTemplates,
+  RenovateConfig,
+} from '../../../config/types';
+import { getManagerList } from '../../../modules/manager';
+import { validMatchFields } from '../../../modules/manager/regex/utils';
+import type { CustomExtractConfig } from '../../../modules/manager/types';
+import type { WorkerExtractConfig } from '../../types';
+
+export interface FingerprintExtractConfig {
+  managerList: Set<string>;
+  managers: WorkerExtractConfig[];
+}
+
+function getRegexManagerFields(
+  config: WorkerExtractConfig
+): CustomExtractConfig {
+  const regexFields = {} as CustomExtractConfig;
+  for (const field of validMatchFields.map(
+    (f) => `${f}Template` as keyof RegexManagerTemplates
+  )) {
+    if (config[field]) {
+      regexFields[field] = config[field];
+    }
+  }
+
+  return {
+    autoReplaceStringTemplate: config.autoReplaceStringTemplate,
+    matchStrings: config.matchStrings,
+    matchStringsStrategy: config.matchStringsStrategy,
+    ...regexFields,
+  };
+}
+
+function getFilteredManagerConfig(
+  config: WorkerExtractConfig
+): WorkerExtractConfig {
+  return {
+    ...(config.manager === 'regex' ? getRegexManagerFields(config) : {}),
+    manager: config.manager,
+    fileMatch: config.fileMatch,
+    npmrc: config.npmrc,
+    npmrcMerge: config.npmrcMerge,
```

Comment:
are those relevant for managers other than npm?

🤔

Reply:
Nope but Rhys didn't want manager-specific filtering to be done here

Reply:
I don't think it will make an *optimization* difference so therefore preferred to keep the code simple and generic.